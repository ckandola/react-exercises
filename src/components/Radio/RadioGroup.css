:root {
    --transition: all 0.3s linear;
}

.radio-form {
    text-align: left;
    margin:auto;
    width:20%;
    background:rgb(71, 143, 0);
    border-radius: 1em;
    padding:0.25em;
    transition: var(--transition);
}

.radio-form:hover {
    box-shadow: 0 5px 15px rgba(0,0,0,0.2);
}

.radio-label {
    color:white;
    margin-left:0.5em;
}

.radio-center-text {
    text-align: center;
}

.radio-submit {
    border-radius: 1em;
    letter-spacing: 2px;
    font-weight: bold;
    border-width: 4px;
    border-color: rgb(68, 56, 143);
    background: rgb(144, 136, 196);
    color:rgb(39, 27, 114);
    padding:0.25em;
    margin-bottom:0.25em;
    transition: var(--transition);
}

.radio-submit:hover {
    background: rgb(68, 56, 143);
    border-color: rgb(144, 136, 196);
    color:rgb(186, 182, 212);
}
