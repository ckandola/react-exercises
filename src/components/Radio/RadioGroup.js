import React, { useState } from 'react';
import './RadioGroup.css';
import PropTypes from 'prop-types';

const RadioGroup = ({name, labels, onSubmit, submitDesc}) => {
    const [selected, setSelected] = useState(null);

    const handleSubmit = e => {
        e.preventDefault();
        onSubmit(selected);
    }

    return (
        <div>
            <h3 className="radio-center-text">Submit to {submitDesc}</h3>
            <form onSubmit={handleSubmit} className="radio-form">
                {labels.map((label, index) => {
                    return (
                        <div key={index}>
                            <input type="radio" name={name} id={index} onClick={() => setSelected(label)}/>
                            <label htmlFor={index} className="radio-label">
                                {label}
                            </label>
                        </div>
                    );
                })}
                <div className="radio-center-text">
                    <button type="submit" className="radio-submit">Submit</button>
                </div>
            </form>
        </div>
    );
};

export default RadioGroup;

RadioGroup.propTypes = {
    name: PropTypes.string,
    labels: PropTypes.arrayOf(PropTypes.string),
    onSubmit: PropTypes.func,
    submitDesc: PropTypes.string
};
