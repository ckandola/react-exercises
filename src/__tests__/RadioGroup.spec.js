import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import RadioGroup from '../components/Radio/RadioGroup';

configure({adapter: new Adapter()});

describe('RadioGroup tests', () => {

    const mockFn = jest.fn();

    const wrapper = shallow(
        <RadioGroup labels={["Credit card", "Paypal", "Check"]} 
            onSubmit={mockFn}
            submitDesc={"test this out."}
        />
    );
    
    it('matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('calls the mock function when Submit is hit', () => {
        wrapper.find(".radio-form").simulate('submit', { preventDefault () {} });
        expect(mockFn.mock.calls).toHaveLength(1);
    });
});
