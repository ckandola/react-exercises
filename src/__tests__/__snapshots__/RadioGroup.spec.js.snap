// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup tests matches the snapshot 1`] = `
<div>
  <h3
    className="radio-center-text"
  >
    Submit to 
    test this out.
  </h3>
  <form
    className="radio-form"
    onSubmit={[Function]}
  >
    <div
      key="0"
    >
      <input
        id={0}
        name="temp"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="radio-label"
        htmlFor={0}
      >
        Credit card
      </label>
    </div>
    <div
      key="1"
    >
      <input
        id={1}
        name="temp"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="radio-label"
        htmlFor={1}
      >
        Paypal
      </label>
    </div>
    <div
      key="2"
    >
      <input
        id={2}
        name="temp"
        onClick={[Function]}
        type="radio"
      />
      <label
        className="radio-label"
        htmlFor={2}
      >
        Check
      </label>
    </div>
    <div
      className="radio-center-text"
    >
      <button
        className="radio-submit"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
